### PR TITLE
Resolve Warnings

### PR DESCRIPTION
**Pull request by ctrimble**
This PR resolves a large number of the warnings found in `org.bridj` and a few warnings in other classes.  Minimal changes were made to the test suite, in an effort to reduce impact to those using the library.

### CODE REVIEW

**Comment by ctrimble** [comment]
Looks like the build failure here is identical to the current failures on master.

File: src/main/java/org/bridj/CallbackNativeImplementer.java [hunk 1]
```diff
@@ -94,17 +98,18 @@ public synchronized ClassDefiner getClassDefiner() {
 
                 Method callbackMethod = runtime.getFastestCallbackMethod(callbackType);
 
-                Class<?>[] parameterTypes = callbackMethod.getParameterTypes();
+                // Never used.  Side effects?
```
Comment (ochafik) [inline src/main/java/org/bridj/CallbackNativeImplementer.java]: Can definitely drop this one, thanks :-)

**Comment by ochafik** [comment]
This is awesome, thanks a lot @ctrimble ! Please tell me you didn't resolve all the warnings by hand, or I'm gonna owe you too many beers for your own health...

**Comment by ctrimble** [comment]
@ochafik the import corrections were totally automated.  The rest was mostly adding wildcard types and taking the default resolution action.

**Pull request merged**